exonum-java-binding-core/rust/src/storage/proof_map_index.rs

```diff
@@ -192,6 +192,35 @@ pub extern "system" fn Java_com_exonum_binding_storage_indices_ProofMapIndexProx
     utils::unwrap_exc_or(&env, res, ptr::null_mut())
 }
 
+/// Returns Java-proof object.
+#[no_mangle]
+pub extern "system" fn Java_com_exonum_binding_storage_indices_ProofMapIndexProxy_nativeGetMultiProof(
+    env: JNIEnv,
+    _: JObject,
+    map_handle: Handle,
+    keys: jbyteArray,
+) -> jobject {
+    let res = panic::catch_unwind(|| {
+        let keys = convert_to_keys(&env, keys)?;
+        let proof = match *utils::cast_handle::<IndexType>(map_handle) {
+            IndexType::SnapshotIndex(ref map) => map.get_multiproof(keys),
+            IndexType::ForkIndex(ref map) => map.get_multiproof(keys),
+        };
+
+        let proof_nodes: JObject = create_java_proof_nodes(&env, &proof)?;
```

Comment:
Sure, sorry I forgot to add `//TODO`. I left it for now in case we decided to get rid of `get_proof` for single proofs at all and only use `get_multiproof` even for single entries. I'll look into implementation of both to decide, whether it makes sense to leave `get_proof` be